File: mobile/native/darwin/Apps/ensu/ensu/Chat/ChatRecovery.swift
import Foundation

enum ChatRecovery {
    private static let resetSignals = [
        "stream pull failed",
        "invalid blob",
        "invalid encrypted"
    ]

    static func shouldResetFromMessage(_ message: String) -> Bool {
        let lowercased = message.lowercased()
        return resetSignals.contains { lowercased.contains($0) }
    }
}

File: mobile/native/android/apps/ensu/data/src/main/java/io/ente/ensu/data/chat/ChatRecovery.kt
package io.ente.ensu.data.chat

internal object ChatRecovery {
    private val resetSignals = listOf(
        "stream pull failed",
        "invalid blob",
        "invalid encrypted"
    )

    fun shouldResetFromMessage(message: String): Boolean {
        val lower = message.lowercase()
        return resetSignals.any { lower.contains(it) }
    }
}
